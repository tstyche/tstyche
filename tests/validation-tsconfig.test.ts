import { afterEach, describe, expect, test } from "@jest/globals";
import { clearFixture, writeFixture } from "./__utils__/fixtureFactory.js";
import { spawnTyche } from "./__utils__/spawnTyche.js";

const isStringTestText = `import { expect, test } from "tstyche";
test("is string?", () => {
  expect<string>().type.toBeString();
});
`;

const fixture = "validation-tsconfig";

afterEach(async () => {
  await clearFixture(fixture);
});

describe("tsconfig", () => {
  test("has errors", async () => {
    const tsconfig = {
      compilerOptions: {
        noEmitOnError: true,
        strict: "yes",
        strictNullChecks: true,
      },
      extends: "../tsconfig.json",
      include: ["./"],
    };

    await writeFixture(fixture, {
      ["__typetests__/dummy.test.ts"]: isStringTestText,
      ["tsconfig.json"]: JSON.stringify(tsconfig, null, 2),
    });

    const { status, stderr, stdout } = spawnTyche(fixture);

    expect(stdout).toMatchSnapshot("stdout");
    expect(stderr).toMatchSnapshot("stderr");

    expect(status).toBe(1);
  });

  test("does not exist", async () => {
    await writeFixture(fixture, {
      ["__typetests__/dummy.test.ts"]: isStringTestText,
    });

    const { status, stderr } = spawnTyche(fixture);

    expect(stderr).toMatch(/^Warning: The default compiler options are used for the following tests files./);

    expect(status).toBe(1);
  });
});
