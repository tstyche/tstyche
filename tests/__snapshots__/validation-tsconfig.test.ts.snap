// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tsconfig has errors: stderr 1`] = `
"Error: Compiler option 'strict' requires a value of type boolean. ts(5024)

  2 |   "compilerOptions": {
  3 |     "noEmitOnError": true,
> 4 |     "strict": "yes",
    |               ^
  5 |     "strictNullChecks": true
  6 |   },
  7 |   "extends": "../tsconfig.json",

      at ./tsconfig.json:4:15

"
`;

exports[`tsconfig has errors: stdout 1`] = `
"uses TypeScript <<version>> with ./tsconfig.json

pass ./__typetests__/dummy.test.ts
  + is string?

Targets:    1 failed, 1 total
Test files: 1 passed, 1 total
Tests:      1 passed, 1 total
Assertions: 1 passed, 1 total
Duration:   <<timestamp>>

Ran all test files.
"
`;
